### DISCUSSION TITLE
Is it possible to activate an addon while creating a new component via API?

### DISCUSSION THREAD

**Discussion by JohnRDOrazio** [WeblateOrg/weblate#11107] [Q&A]
I have started working recently with the weblate API in order to automatically create project components from InDesign documents. I was just wondering, when creating a component using `fetch` with `POST` method to `https://{path-to-weblate-api}/projects/{project-slug}/components/`, is it possible to enable addons at the same time? Or does this have to be done in a successive API call?

I have successfully activated an addon with a successive API call to the `https://{path-to-weblate-api}/components/{project-slug}/{component-slug}/addons/` endpoint using `POST` method with a value like this:

```js
{
  name: "weblate.autotranslate.autotranslate",
  configuration: {
    mode: "suggest",
    engines: "weblate",
    component: null,
    threshold: 80,
    auto_source: 'others',
    filter_type: 'todo'
  }
}
```
and I see that the addon is successfully activated.

I have attempted to do the addon activation at time of component creation by adding an `addons` key to the JSON payload with the same value as above wrapped in an array: `addons: [ { name: "weblate.autotranslate.autotranslate", configuration: { ... } } ]`. However this does not seem to work. Is it possible to activate addons during component creation in a single call?

**Comment by nijel** [comment]
No, add-ons have to be installed in an additional API call. If this is something you often do, you might want to use https://docs.weblate.org/en/latest/admin/config.html#default-addons.

PS: The upcoming release will bring possibilty to install add-ons on project or site scopes as well:

* https://github.com/WeblateOrg/weblate/issues/10052 
* https://github.com/WeblateOrg/weblate/issues/10051

**Reply by eheck-is** [reply]
Is there a way to find out which add-ons will work project-wide or side-wide execpt try-and-error? I've installed the autotranslate add-on to a project and it doesn't do anything. Missing strings get translated only when I add it to the individual components. I have version 5.7.2 (I still have Python 3.10 and cannot update right now)

**Reply by nijel** [reply]
The only add-on which has limited scopes is component discovery, any other add-on should work in any scope. I know we fixed some issues like that, but I don't recall if this is something that was fixed in the 5.8 release.
